File: Javascript/Array/66.加一.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
/**
 * 遍历数组给最后一个元素加一
 * 若果最后一个数组为9 则需要给前一个元素加一
 * 如果最后一个元素不为9 直接给最后一个元素加一
 * 如果第一个元素为9，则需要给数组头添加1
 */
var plusOne = function(digits) {
    for(let i=digits.length-1;i>= 0;i--) {
        if(digits[i] == 9) {
            digits[i] = 0 
            if(i == 0) {
                digits.splice(0, 0, 1)
                return digits
            }
        }
        else{
            digits[i]++
            return digits
        }
    }
    
};

File: Go/ArrayList/66-加一.go
package ArrayList

func plusOne(digits []int) []int {
	f := 1
	for i := len(digits)-1; i >= 0; {
		if digits[i]+f == 10 {
			digits[i] = 0
			i--
		} else {
			// 不需要进位的，加法结束，直接返回
			digits[i] += 1
			return digits
		}
	}
	// 在头部插入1
	digits = append([]int{1}, digits...)
	return digits
}
